src/SfxWeb/src/app/views/replica/commands/commands.component.ts

```diff
@@ -0,0 +1,118 @@
+import { Component, Injector } from '@angular/core';
+import { Observable } from 'rxjs';
+import { IResponseMessageHandler } from 'src/app/Common/ResponseMessageHandlers';
+import { CommandFactory, CommandParamTypes, CommandSafetyLevel, PowershellCommand, PowershellCommandParameter } from 'src/app/Models/PowershellCommand';
+import { DataService } from 'src/app/services/data.service';
+import { ReplicaBaseControllerDirective } from '../ReplicaBase';
+
+@Component({
+  selector: 'app-commands',
+  templateUrl: './commands.component.html',
+  styleUrls: ['./commands.component.scss']
+})
+export class CommandsComponent extends ReplicaBaseControllerDirective{
+  
+  commands: PowershellCommand[] = [];
+
+  constructor(protected data: DataService, injector: Injector) {
+    super(data, injector);
+  }
+
+  refresh(messageHandler?: IResponseMessageHandler): Observable<any> {
+    return this.data.getNodes();
+  }
+
+  afterDataSet(): void {
+    this.setUpCommands();
+  }
+
+  private setUpCommands() {
```

Comment:
this might need to be ran on refresh because there is a possibility that replica roles could change.